add support for Darwin platform

frepond opened the pull request:
Hi Austin, we have some guys here using MacOs for development. I've just added a minor modification in order to use the right binary depending on the platform (Linux or Darwin).

frepond committed:
add support for Darwin platform

AustinScola commented:
Hey Federico, thanks, I was aware of this limitation but didn't document it yet. The repo in general is based off of [nodkz/mongodb-memory-server ](https://github.com/nodkz/mongodb-memory-server) which does a great job of figuring out which binary to download. It might provide a good model for further support.

I currently have a test coverage requirement of 100% so I think you will probably have to write a unit test for this. Also, you'll have to forgive me but I am struggling to figure out how to approve the workflow jobs on this PR to run?

AustinScola committed:
Fix Python workflow triggers

When @frepond opened PR #21, I wasn't able to figure out how to run the
Python workflow jobs which are required for merging. At first I thought
it was because I couldn't find the button to approve them. But I think
it is actually because the triggers for the job were to restrictive.

AustinScola committed:
Restrict Python workflow triggers by branch

AustinScola commented on pytest_motor/plugin.py:
```diff
@@ -33,21 +34,21 @@ async def root_directory(pytestconfig: PytestConfig) -> Path:
 
     if not mongod_binary_filename.exists():
-        download_url = 'https://fastdl.mongodb.org/linux/mongodb-linux-x86_64-ubuntu1804-4.4.6.tgz'
+        download_url = f'https://fastdl.mongodb.org/osx/{__mongo_exec()}.tgz'
```
Is `linux/` needed for this to still work on Ubuntu?

frepond replied:
Yes, that's wrong it varies depending on the platform. I'm fixing it.

frepond committed:
Merge branch 'AustinScola:master' into master

frepond committed:
fixed scripts and new tests
- fixed scripts and working path for macos
- new tests
- refactor fun name

AustinScola commented on pytest_motor/plugin.py:
```diff
@@ -27,27 +28,26 @@ def event_loop() -> Iterator[asyncio.AbstractEventLoop]:
-async def root_directory(pytestconfig: PytestConfig) -> Path:
-    """Return the root path of pytest."""
-    return pytestconfig.rootpath
+async def root_directory() -> Path:
+    """Return the working path."""
+    return Path(tempfile.gettempdir())
```
I don't think we want to do this because it will mean that the binary has to be downloaded every run. The reason for using a fixed path is that we can check if the mongod binary has already been downloaded, and then only download it if it doesn't exist.

frepond replied:
Ok, it's wip. Let me think a better alternative that works on MacOs. There still a few things to fix. I wasnt aware that every push is generating a PR.

frepond replied:
@AustinScola  just thinking, what if the location is configured via env variable? and defaults to `pytestconfig.rootpath` if not present?

AustinScola replied:
Sounds good to me

AustinScola replied:
Is the problem that the `pytestconfig.rootpath` doesn't work or is wrong on MacOS?

frepond replied:
Sorry for the delay, very busy these days.

`pytestconfig.rootpath` value seems to be ok. It starts mongod apparently but then fails to connect the socket. With a modification, I'm pushing you'll be able to override the value with `PYTEST_MOTOR_PATH=/tmp ./scripts/all.sh`.

AustinScola replied:
What is the reason for failing to connect to the socket?

frepond replied:
With your recent changes, it seems to work fine on MacOs. Regarding `PYTEST_MOTOR_PATH ` it could be still useful in some cases. I have a bunch of microservices and downloading the mongo binaries for each of them is cumbersome. Also `readlink -f` is working. You have to install `greadlink` (GNU) and create an alias.

AustinScola commented on scripts/all.sh:
```diff
@@ -2,7 +2,7 @@
 set -eu
 
-HERE="$(dirname "$(readlink -f "$BASH_SOURCE")")"
+HERE="$( cd -- "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )"
```
I'm a bit of a bash noob, would you mind explaining what this change does for me?

frepond replied:
Basically the same, getting the script dir, `readlink -f` doesn't work on MacOs.

frepond committed:
fixed linux download url and optinal env var path
- download url had "osx" harcoded preventing it to work on linux
- override "pytestconfig.rootpath" as working dir with env var PYTEST_MOTOR_PATH

AustinScola commented:
@frepond I think the merge conflicts have to be resolved before I can approve the GitHub Action workflows to run?

aleksul commented:
This PR should be closed because of PR #39, I guess.
